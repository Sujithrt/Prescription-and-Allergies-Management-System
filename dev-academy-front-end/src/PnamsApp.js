import './App.css';
import {
    BrowserRouter,
    Switch,
    Route,
} from "react-router-dom";
import LoginPage from './pages/loginPage/LoginPage';
import Dashboard from './pages/dashboard/Dashboard';
import UserInfo from './pages/userInfo/UserInfo';
import PatientProfile from './pages/patientProfile/PatientProfile';
import AuthenticatedRoute from './components/AuthenticatedRoute';
import ErrorComponent from './pages/errorComponent/ErrorComponent';

/**
 * Manages Routing between different pages of the Applicaiton
 * 
 */
function PnamsApp() {
    return (
        <BrowserRouter>
            <Switch>
                <Route exact path="/" component={LoginPage} />
                <Route exact path="/login" component={LoginPage} />
                <AuthenticatedRoute exact path="/dashboard" component={Dashboard} />
                <AuthenticatedRoute exact path="/user" component={UserInfo} />
                <AuthenticatedRoute exact path="/patient/:id" component={PatientProfile} />
                <Route path="" component={ErrorComponent} />
                <Route path="/error" component={ErrorComponent} />
            </Switch>
        </BrowserRouter>
    );
}

export default PnamsApp;
