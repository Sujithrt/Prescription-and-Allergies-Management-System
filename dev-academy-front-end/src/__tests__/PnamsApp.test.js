import React from 'react';
import { shallowToJson } from "enzyme-to-json";
import { shallow } from 'enzyme';
import PnamsApp from './../PnamsApp';

describe("PnamsApp Test", ()=>{
  it("PnamsApp is rendered as expected", ()=>{
    const pnamsApp = shallow(<PnamsApp />);
    expect(shallowToJson(pnamsApp)).toMatchSnapshot();
  });
});