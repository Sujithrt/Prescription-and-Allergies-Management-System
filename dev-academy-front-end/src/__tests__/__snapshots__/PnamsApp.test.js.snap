// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PnamsApp Test PnamsApp is rendered as expected 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <AuthenticatedRoute
      component={[Function]}
      exact={true}
      path="/dashboard"
    />
    <AuthenticatedRoute
      component={[Function]}
      exact={true}
      path="/user"
    />
    <AuthenticatedRoute
      component={[Function]}
      exact={true}
      path="/patient/:id"
    />
    <Route
      component={[Function]}
      path=""
    />
  </Switch>
</BrowserRouter>
`;
